common/plugins/eu.esdihumboldt.hale.common.core/src/eu/esdihumboldt/hale/common/core/io/IOProvider.java

```diff
@@ -62,6 +62,21 @@
 	public IOReport execute(ProgressIndicator progress)
 			throws IOProviderConfigurationException, IOException;
 
+	/**
+	 * Execute the I/O provider.
+	 * 
+	 * @param progress the progress indicator, may be <code>null</code>
+	 * @param resourceIdentifier identifier of the provided resource. Null if a
+	 *            new resource id needs to be generated.
+	 * @return the execution report
+	 * 
+	 * @throws IOProviderConfigurationException if the I/O provider was not
+	 *             configured properly
+	 * @throws IOException if an I/O operation fails
+	 */
+	public IOReport execute(ProgressIndicator progress, String resourceIdentifier)
```

Comment:
From the usage and the number of implementations that just return null in the implementation I think the `IOProvider` interface is not the right location to add this method.
Maybe instead in `ImportProvider`?

This seems to be a method that is only called in a special case. It would be good if the method documentation goes into a bit more detail on that.

Reply:
Method moved to `ImportProvider` and updated the java doc.